TEST_APPS/testcases/netsocket/README.md

```diff
@@ -0,0 +1,7 @@
+Netsocket tests
+===============
+
+This folder contains netsocket tests for Icetea
+The tests located under this folder are dependent of the application [socket_app](https://github.com/ARMmbed/mbed-os/blob/master/TEST_APPS/device/socket_app)
+
+The test cases under this folder are defined in [Network Socket test plan](https://github.com/ARMmbed/mbed-os/blob/master/TESTS/netsocket/README.md)
```

Comment:
@SeppoTakalo , can you help us to review this please?
The "Network Socket test plan" does not have references to Icetea, so I don't see the relationship with doc.

Reply:
As a user, I still don't understand how to run the Icetea test cases with the network sockets.
@jarlamsa @SeppoTakalo can we add a line explaining how to run the test cases?

We have this today (greentea):
`mbed test --compile -t <toolchain> -m <target> -n mbed-os-tests-network-*,mbed-os-tests-netsocket*`

I *assume* this is the corresponding command to use icetea but I'm not 100% sure.

`mbed test --compile -t <toolchain> -m <target> -n mbed-os-tests-network-*,mbed-os-tests-netsocket* --icetea`

Reply:
There is already a readme about running the icetea tests in the root of TEST_APPS -folder
https://github.com/ARMmbed/mbed-os/blob/master/TEST_APPS/readme.md

Reply:
Ok, that's good.
@SeppoTakalo , I assume every test case (`mbed-os-tests-network-*,mbed-os-tests-netsocket*`) can be used with Icetea. If that's the case, I'd be happy to close this item.

Reply:
Only test cases done with Icetea can be used with Icetea, those are located under TEST_APPS/testcases, test cases done with Greentea can be used with Greentea, those are located under TESTS. The test plan needs information about used test framework for each test.

Reply:
Yes.
Some testcases exist only in Greentea, some exist only in Icetea.
All testcases exist in the test plan, and the selection of tool is based on requirement of the testcase.

For demonstration purposes some testcases exist on both, so that the differences can be studied.